### PR TITLE
Add some :throws: annotations in standard modules

### PR DESCRIPTION
**Pull request by vasslitvinov**
While there, do some other cleanups,
such as removing claims that a function halts upon error.

### CODE REVIEW

File: modules/standard/FileSystem.chpl [hunk 1]
```diff
@@ -265,8 +263,8 @@ proc chown(out error: syserr, name: string, uid: int, gid: int) {
 // When basename and joinPath are supported, enable dest to be a directory.
 
 /* Copies the contents and permissions of the file indicated by `src` into
-   the file or directory `dest`.  If `dest` is a directory, will throw a
-   FileNotFoundError.  If `metadata` is set to `true`, will also copy the
+   the file or directory `dest`.
+   If `metadata` is set to `true`, will also copy the
```
Comment (lydia-duncan) [inline modules/standard/FileSystem.chpl]: The line wrapping after removing that information is weird, should be fixable with a quick Meta-Q in emacs

File: modules/standard/IO.chpl [hunk 2]
```diff
@@ -315,14 +315,18 @@ Error Handling
 Most I/O routines throw a :class:`SysError.SystemError`, and can be handled
 appropriately with ``try`` and ``catch``. For legacy reasons most I/O routines
 can also can accept an optional `error=` argument.
+In this documentation, `SystemError` may include both the
```
Comment (lydia-duncan) [inline modules/standard/IO.chpl]: I think this should be two ` instead of just one on each side

File: modules/standard/IO.chpl [hunk 3]
```diff
@@ -609,10 +612,12 @@ proc stringStyleWithVariableLength() {
 }
 
 /*
-  This method returns the appropriate :record:`iostyle` ``str_style`` value
-  to indicate a string format where string data is preceded by a ``lengthBytes``
-  of length. Only lengths of 1, 2, 4, or 8 are supported; if this method
-  is called with any other length, it will throw an error.
+  Return the appropriate :record:`iostyle` ``str_style`` value
+  to indicate a string format where string data is preceded by a `lengthBytes`
```
Comment (lydia-duncan) [inline modules/standard/IO.chpl]: Why did you remove the double ` here?

File: modules/standard/IO.chpl [hunk 4]
```diff
@@ -6358,10 +6375,12 @@ proc channel._read_complex(width:uint(32), out t:complex, i:int)
    Write arguments according to a format string. See
    :ref:`about-io-formatted-io`.
 
-   Throws a SystemError if the arguments could not be written.
-
    :arg fmt: the format string
-   :arg args: the arguments to write
+   :arg args: 0 or more the arguments to write
```
Comment (lydia-duncan) [inline modules/standard/IO.chpl]: "0 or more the arguments" -> "0 or more arguments"

File: modules/standard/IO.chpl [hunk 5]
```diff
@@ -6527,7 +6546,7 @@ proc channel.writef(fmtStr:string, const args ...?k, out error:syserr):bool {
   return false;
 }
 
-// documented in varargs version
+pragma "no doc" // documented in varargs version
```
Comment (lydia-duncan) [inline modules/standard/IO.chpl]: Question about this and the other thing you "no doc"d: are the argument lists sufficiently different to justify both being in the documentation, or could one have been inferred from the other?  I might wait to check with @mppf on those changes

File: modules/standard/IO.chpl [hunk 6]
```diff
@@ -6944,7 +6962,7 @@ proc writef(fmt:string):bool {
     return stdout.writef(fmt);
   }
 }
-/* Call ``stdout.readf``; see :proc:`channel.readf`. */
+/* Call ``stdin.readf``; see :proc:`channel.readf`. */
```
Comment (lydia-duncan) [inline modules/standard/IO.chpl]: D'oh!

File: modules/standard/SysError.chpl [hunk 7]
```diff
@@ -391,8 +391,8 @@ private proc quote_string(s:string, len:ssize_t) {
 
    :arg error: the error code
    :arg msg: extra information to include in the thrown error
-   :arg path: a path to include in the thrown error
-   :arg offset: an offset to include in the thrown error
+   :arg path: optionally, a path to include in the thrown error
+   :arg offset: optionally, an offset to include in the thrown error
```
Comment (lydia-duncan) [inline modules/standard/SysError.chpl]: I think it would be fine to keep this change in though

File: modules/standard/IO.chpl (hunk 3 above)
Comment (vasslitvinov) [reply modules/standard/IO.chpl]: We tend to put the names of formals in single backticks.
